app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -10,6 +10,25 @@ def auth_methods_session
     @auth_methods_session ||= AuthMethodsSession.new(user_session:)
   end
 
+  def handle_valid_verification_for_authentication_context(auth_method:)
+    mark_user_session_authenticated(auth_method:, authentication_type: :valid_2fa)
+    disavowal_event, disavowal_token = create_user_event_with_disavowal(:sign_in_after_2fa)
+
+    if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
+      if current_user.sign_in_new_device_at.blank?
+        current_user.update(sign_in_new_device_at: disavowal_event.created_at)
+      end
```

Comment:
This regressed in cf33caee521be4bcecb38c9b638824f38079b093 to sending a notification on every sign-in, even for an existing device. We should add some test coverage to protect against this regression.

Possible fixes:

- Assign `sign_in_new_device_at` from PIV/CAC handler before calling this method (complexity in needing to keep event date and `sign_in_new_device_at` in sync)
- Handle `PIV/CAC` explicitly here (complexity in handling "passwordless" PIV/CAC vs. PIV/CAC as MFA method)
- Add condition to check if this is a new device here (e.g. `User#new_device?`)

Reply:
Regression test added in d449b1b. Fixed in 54b62f4 using the third option listed in previous comment.